docs/architecture/2-searchBank.md

```diff
@@ -55,13 +88,58 @@ Step 8. through 11. is repeated as long as PSU modifies keywords (by adding or r
 
 Like this diagram shows, local bank search does not send request to the network.
 
+#### LocalSearch-001 FinTechUI.loadFinTechSeachScreen
+PSU loads the FinTechSearchString of the FinTechUI.
+
+#### LocalSearch-101, -103 FinTechApi.bankSearch
+The FinTechUI sends a get request to the FinTechApi. If the List<BankProfile> is not yet loaded by the FinTechUI, the FinTechUI issues a bankSearch request to the FinTechApi. A keyword passed as query parameter allows the backend to reduce the initial extent of records included in the index.
+
+
+* keyword (searchInput): The bank search input string. Must be used to reduce the extent of records to be included in the local search. E.g.: c=DE could limit the search to bank in Germany.
+* start (searchStartIndex): The index of the first result. The start will be set to 0 to get all records.
+* max (searchMaxResult): the max number of entries to return with the response. Set this number to -1 to receive all applicable records from the server.
```

Comment:
Given bank search API is more or less open to public, allowing `max=-1` to return all results (it might be 1000+ banks) may heavily affect performance or introduce DoS. Typically API's limit max to some reasonable maximum

Reply:
This is a permissioned API. And tpp can cache results...

Reply:
Who really needs 100+ matching banks as response result, typically you need only 10 or so to show in UI.
Ones who want to cache stuff - can go page by page to build their cache.
All large API providers do limit maximum response size. I do not see why we should send 1Mb of data as a response to somebody even if he has rights

Reply:
Hi Francis, I agree with Val. But I would define a limit to max at 100 or less too. However, this could be designed later. For now I think we have to merge all important branches to develop asap.